draft-ietf-rtcweb-security-arch.xml

```diff
@@ -556,6 +556,158 @@
+  identity-assertion-value  = base64
+  identity-extension        = extension-name [ "=" extension-value ]
+  extension-name            = token
+  extension-value           = 1*(%x01-09 / %x0b-0c / %x0e-3a / %x3c-ff)
+                              ; byte-string from [RFC4566]
+
+  <ALPHA and DIGIT as defined in [RFC4566]>
+  <base64 as defined in [RFC4566]>
+  
+ Example:
+
+  a=identity:\
+    eyJpZHAiOnsiZG9tYWluIjoiZXhhbXBsZS5vcmciLCJwcm90b2NvbCI6ImJvZ3Vz\
+    In0sImFzc2VydGlvbiI6IntcImlkZW50aXR5XCI6XCJib2JAZXhhbXBsZS5vcmdc\
+    IixcImNvbnRlbnRzXCI6XCJhYmNkZWZnaGlqa2xtbm9wcXJzdHV2d3l6XCIsXCJz\
+    aWduYXR1cmVcIjpcIjAxMDIwMzA0MDUwNlwifSJ9
+    
+  Note that long lines in the example are folded to meet the column
+  width constraints of this document; the backslash ("\") at the end of
+  a line and the carriage return that follows shall be ignored.
+  
+      ]]></artwork>
+       </figure>
+       <t>
+         This specification does not define any extensions for the attribute.
+       </t>
+       <t>
+         The identity-assertion value is a JSON <xref target="RFC8259"/> encoded string. The JSON object 
+         contains two keys: "assertion" and "idp". The <spanx style="verb">assertion</spanx> key value contains 
+         an opaque string that is consumed by the IdP. The <spanx style="verb">idp</spanx> key value contains a
+         dictionary with one or two further values that identify the IdP. See
+         <xref target="sec.request-assert"/> for more details.
+       </t>
+       <section title="Offer/Answer Considerations" anchor="sec.sdp-id-attr-oa">
+         <t>
+		   This section defines the SDP Offer/Answer <xref target="RFC6454"/> considerations for the SDP
+		   'identity' attribute.
+	     </t>
+         <t>
+           Within this section, 'initial offer' refers to the first offer in the
+           SDP session that contains an SDP <spanx style="verb">identity</spanx> attribute.
+         </t>
+         <section title="Generating the Initial SDP Offer" anchor="sec.sdp-id-attr-oa-inio">
+         <t>
+           When an offerer sends an offer, in order to provide its identity
+           assertion to the peer, it includes an 'identity' attribute in the
+           offer. In addition, it MUST include SDP 'fingerprint' attributes for
+           each fingerprint that has been used to assert the identity.
+         </t>
+         </section>
+         <section title="Answerer Processing of SDP Offer" anchor="sec.sdp-id-attr-oa-anso">
+         <t>
+           When an answerer receives an offer that contains an 'identity'
+           attribute, the answerer can use the the attribute information to
+           contact the IdP, and verify the identity of the peer.  If the
+           answerer does not accept the attribute value, or if the identity
+           verification fails, it MUST reject the offer.
+         </t>
+         </section>
+         <section title="Generating of SDP Answer" anchor="sec.sdp-id-attr-oa-ansa">
```

Comment:
Please don't duplicate these sections. I'd prefer that we just have two sections ("generation" and "processing") but if you insist on having four, then the answer ones should just point to the offer ones. Duplication is obfuscatory.

Reply:
Eventhough the text is identical, I'd like to keep it. Note that I previously did remove these sections from the "Modifying the session" section.

Reply:
This is an editor decision, but if you prefer, I'm happy to merge the PR and then remove it myself

Reply:
Let me think about it. I have not pushed a new commit that should address your other comments.